src/regtech_data_validator/data_formatters.py

```diff
@@ -104,53 +102,54 @@ def df_to_table(df: pd.DataFrame) -> str:
 
 
 def df_to_json(df: pd.DataFrame) -> str:
-    output_json = []
-
-    if not df.empty:
-        df.reset_index(drop=True, inplace=True)
-        findings_json = json.loads(df.to_json(orient='columns'))
-
-        grouped_data = {}
-        for i in range(len(findings_json['record_no'])):
-            validation_id = findings_json['validation_id'][str(i)]
-            if validation_id not in grouped_data:
-                grouped_data[validation_id] = []
-            grouped_data[validation_id].append(
-                {
-                    'record_no': findings_json['record_no'][str(i)],
-                    'uid': findings_json['uid'][str(i)],
-                    'field_name': findings_json['field_name'][str(i)],
-                    'field_value': findings_json['field_value'][str(i)],
-                }
-            )
-
-        for validation_id, records in grouped_data.items():
-            for key, value in findings_json['validation_id'].items():
-                if validation_id == value:
-                    validation_key = key
-                    break
-            validation_info = {
-                'validation': {
-                    'id': validation_id,
-                    'name': findings_json['validation_name'][validation_key],
-                    'description': findings_json['validation_desc'][validation_key],
-                    'severity': findings_json['validation_severity'][validation_key],
-                    'scope': findings_json['scope'][validation_key],
-                    'fig_link': findings_json['fig_link'][validation_key],
-                },
-                'records': [],
+    # grouping and processing keeps the process from crashing on really large error
+    # dataframes (millions of errors).  We can't chunk because could cause splitting
+    # related validation data across chunks, without having to add extra processing
+    # for tying those objects back together.  Grouping adds a little more processing
+    # time for smaller datasets but keeps really larger ones from crashing.
```

Comment:
when you say crashing, does it kill the whole API with OOM or something like that? Or just the validation step fails? While "chunking" is worthwhile regardless, if just the validation fails, and doesn't kill the pod, then I'm less concerned with validation; maybe post-mvp we can add some sort of file has too many errors to process type of thing.

Reply:
https://github.com/cfpb/sbl-filing-api/pull/224 was written to handle if an executor crash happens during processing in the filing-api and gracefully move the submission to VALIDATION_ERROR